packages/grid/_modules_/grid/hooks/features/virtualization/useVirtualColumns.ts

```diff
@@ -39,6 +39,8 @@ export const useVirtualColumns = (
   const columnsMeta = useGridSelector(apiRef, columnsMetaSelector);
   const visibleColumnCount = useGridSelector(apiRef, visibleColumnsLengthSelector);
   const visibleColumns = useGridSelector(apiRef, visibleColumnsSelector);
+  const getLastVisibleColIndex = (columns) => (columns.length > 0 ? columns.length - 1 : 0);
```

Comment:
Either move this one outside the component or useCallback, because atm you recreate it all the time which makes the further useCallback useless. Also I believe you need to update a dependency array further down

Reply:
I cleaned it up, that is no longer needed.